### ISSUE TITLE
Allow symbolic indices to be passed to the `save_idxs` kwarg of `solve`

### ISSUE THREAD

**Issue by jonathanfischer97** [SciML/DifferentialEquations.jl#1036]
## Can't use symbolic indices to parameterize `solve(;save_idxs = ...)`

- Often I want to just save an **observed** variable when solving an `ODEProblem`, to save on memory during large batch solves.
- Currently that isn't possible because the observed variable is rightfully not included within `u`, but computed lazily via the solution.
  - Resorted to defining my own tedious `save_func(u, t, integrator)` callback, forcing me to redefine my observable equations in terms of `u[idx]`.
- If I still want to symbolically index the `ODESolution` for my observed variable, I have to save all of the variables during `solve`, which is a lot of waste for larger systems.

### Would be nice to be able to pass in symbols to `solve` to be saved, like `solve(odeprob, alg; save_idxs = [:obs, ...]`

Basically, I'd like `solve` to handle symbolic indices from an `ODESystem` just like `remake` and solution indexing already do.

**Comment by ChrisRackauckas** [comment]
Yes this is a current known limitation of the symbolic indexing. It will take a bit of work to fix but it's in the plans.

**Comment by ChrisRackauckas-Claude** [comment]
Retested during a triage sweep — this is **not** a duplicate of #814, and it is still unimplemented. Flagging because the two look similar enough to be closed together by mistake.

#814 (symbolic *state* names) now works. The ask here — saving **observed** variables — is explicitly rejected:

```
save_idxs = obs   -> ArgumentError: Can only save variables and timeseries parameters. Got obs(t).
SII.is_variable(prob, obs) = false
SII.is_observed(prob, obs) = true
```

That rejection is deliberate, at `SciMLBase/src/solutions/save_idxs.jl:227`. The 2024 comment in this thread describing it as a known limitation still stands, so this should stay open.

Verified on DifferentialEquations v8.0.2 / OrdinaryDiffEq v7.1.3 / Julia 1.11.9.

**Comment by ChrisRackauckas-Claude** [comment]
## Reconfirmed (decision triage)

Still a **feature request only** — not a half-implemented crash path.

| Concern | Status |
|---|---|
| Symbolic **state** `save_idxs` (#814) | Implemented via `SavedSubsystem` / `get_save_idxs_and_saved_subsystem` |
| Symbolic **observed** `save_idxs` (this issue) | Explicitly rejected: `ArgumentError: Can only save variables and timeseries parameters` at `SciMLBase/src/solutions/save_idxs.jl` |
| Partial observed work | Solution *indexing* of observed after full solve works; save-time selection of observed does not |

No draft PR: supporting observed `save_idxs` needs a designed mechanism (compute observed during save without keeping full `u`, or force full-state save + thin solution interface). Out of scope for a small unambiguous fix.

Keep open; distinct from #814.

**Comment by ChrisRackauckas-Claude** [comment]
## Design sketch (feature still unimplemented)

Reconfirmed on ModelingToolkit v11.36 / SciMLBase v3.37 / OrdinaryDiffEq v7.1.3:

```
is_variable(prob, obs)  = false
is_observed(prob, obs)  = true
solve(...; save_idxs=[obs])
  → ArgumentError: Can only save variables and timeseries parameters. Got obs(t).
```

Rejection is deliberate in `SciMLBase/src/solutions/save_idxs.jl` (`SavedSubsystem` constructor): only `variable_index` and `timeseries_parameter_index` are accepted. Full-solve `sol[obs]` still works via the observed function.

Distinct from symbolic **state** `save_idxs` (DifferentialEquations.jl#814 / SciMLBase SavedSubsystem path) — that path is done.

### Why this is not a small patch

Saving an observed without keeping full `u` means either:

1. **Eval-at-save**: at each save point compute `obs(u,p,t)` and store only that scalar/vector in `sol.u`, while still answering `sol[obs]` / interpolation, **or**
2. **Save full state, thin interface**: keep full `u` internally (defeats memory goal) and only *expose* observed through the solution API.

(1) is what users want for batch memory. It needs design choices:

| Surface | Notes |
|---|---|
| `save_idxs = [obs, x, …]` mixed | Observed + true states in one solution; `SavedSubsystem` maps must distinguish “stored state index” vs “stored observed column” |
| Interpolation | Dense output for pure observed may require either storing enough state history to re-eval `observed`, or only supporting non-dense / `saveat` for observed-only solves |
| `sol[x]` after observed-only save | Must error clearly if `x` was not saved and cannot be reconstructed |
| Callbacks / events / adjoints | Today assume full or subset **state**; observed-only buffers need a contract |
| Discrete / parameter timeseries | Already partially handled; observed is a third category |

### Suggested implementation sketch (SciMLBase)

1. Extend `SavedSubsystem` (or sibling) with `observed_map`: saved-column → observed symbolic + `observed(sys, sym)` callable.
2. In the integrator save path, if `is_observed(indp, sym)`, evaluate observed and push to the save buffer (not `u[idxs]`).
3. Solution indexing: `getsym` uses the observed column when present; refuse dense interp of observed unless parent state was also saved (v1 policy).
4. Document: `save_idxs` observed support requires `saveat` / non-dense for v1; full dense observed-only is a later phase.

### Docs note

No wrong docs found claiming observed `save_idxs` already works; the kwarg docs describe state indices / symbolic states. No docs PR needed.

**No draft PR** — pure feature with non-trivial save/interp design. Keep open; not a duplicate of #814.

**Comment by ChrisRackauckas-Claude** [comment]
## Smallest correct slice landed (error + docs; not full feature)

**Draft PR:** https://github.com/SciML/SciMLBase.jl/pull/1467

### Reconfirmed failure mode

```julia
@mtkcompile sys = System([D(x) ~ …, D(y) ~ …, obs ~ x + y], t)
solve(prob; save_idxs = [obs])
# before: ArgumentError: Can only save variables and timeseries parameters. Got obs(t).
# after:  ArgumentError: Saving observed variables via `save_idxs` is not yet supported …
#         workarounds + DifferentialEquations.jl#1036
```

`is_variable(prob, obs) = false`, `is_observed(prob, obs) = true`. Rejection was always deliberate at `SavedSubsystem` construction; the message did not name *observed*.

### Why not implement full support here

| Path | Verdict |
|---|---|
| Eval-at-save observed columns | Correct target for memory savings, but needs `SavedSubsystem.observed_map` **and** OrdinaryDiffEqCore save path changes (`u[save_idxs]` today) + interpolation policy |
| Silently save full `u` when observed requested | Works for `sol[obs]` but **not** memory-saving — half-broken silent wrong answer |
| SciMLBase-only change under ~100 LOC that actually stores observed | Not available; architecture does not half-support it |

### This PR

- Dedicated `ArgumentError` when `is_observed` for invalid `save_idxs` entries
- Docs note on symbolic save_idxs + common keywords
- Unit + MTK downstream tests

### Implementation sketch for the real feature (unchanged from prior design)

1. `SavedSubsystem` + `observed_map`: saved-column → observed symbolic + `observed(sys, sym)` callable
2. Integrator save path: if observed, evaluate at save points into buffer (not `u[idxs]`)
3. Solution indexing: `getsym` reads observed column; refuse dense interp of pure observed unless parent state also saved (v1)
4. Document: observed `save_idxs` v1 requires `saveat` / non-dense

Keep open until that lands. Distinct from #814 / SciMLBase #1460 (symbolic **state** `save_idxs`).
